src/Microsoft.AspNetCore.Blazor/Components/TreeParameterState.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Blazor.Rendering;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    internal readonly struct TreeParameterState
+    {
+        private readonly static IDictionary<Type, ReflectedTreeParameterInfo[]> _cachedInfos
```

Comment:
nit: don't use `IDictionary<>` here. This will be more efficient, and concurrent dictionary exposes lots of API that `IDictionary<>` doesn't.

Reply:
Good point. Done.